src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -76,6 +80,15 @@ upgrade_to_reference_table(PG_FUNCTION_ARGS)
 								  relationName)));
 	}
 
+	if (tableEntry->replicationModel == REPLICATION_MODEL_STREAMING)
+	{
+		char *relationName = get_rel_name(relationId);
+		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
```

Comment:
Suggestion for the message : `cannot upgrade to reference table because of replication model mismatch`

Suggestion for the detail: `Upgrade is only supported for relations that are replicated using statement-based replication where "%s" is replicated using streaming replication`